### PR TITLE
gpgme: Fix build with python on macOS

### LINKED ISSUE #70996: darwin: gpgme build broken
While reviewing #70990 I noticed, the bug on a downstream dependency

**Describe the bug**
tries to do a dlopen, and can't find the dynamic library.

**Expected behavior**
Not crash.

**Additional context**
<details> <summary> Fail log </summary>

```
Traceback (most recent call last):
  File "./final.py", line 24, in <module>
    import support
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/tests/support.py", line 27, in <module>
    import gpg
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/__init__.py", line 123, in <module>
    from . import core
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/core.py", line 10, in <module>
    from . import gpgme
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/gpgme.py", line 28, in <module>
    _gpgme = swig_import_helper()
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/gpgme.py", line 24, in swig_import_helper
    _mod = imp.load_module('_gpgme', fp, pathname, description)
  File "/nix/store/611gq7s9s11w5ri95fbq2adi9ifrhkmn-python3-3.7.4/lib/python3.7/imp.py", line 242, in load_module
    return load_dynamic(name, filename, file)
  File "/nix/store/611gq7s9s11w5ri95fbq2adi9ifrhkmn-python3-3.7.4/lib/python3.7/imp.py", line 342, in load_dynamic
    return _load(spec)
ImportError: dlopen(/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/_gpgme.cpython-37m-darwin.so, 2): Library not loaded: /nix/store/95ksx1ywkpp1id8k3xqp9x219r0fjj0k-gpgme-1.13.1/lib/libgpgme.11.dylib
  Referenced from: /private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/_gpgme.cpython-37m-darwin.so
  Reason: image not found
Running tests using /nix/store/611gq7s9s11w5ri95fbq2adi9ifrhkmn-python3-3.7.4/bin/python3.7 (3.7)...
FAIL: initial.py
FAIL: t-wrapper.py
FAIL: t-callbacks.py
FAIL: t-data.py
FAIL: t-encrypt.py
FAIL: t-encrypt-sym.py
FAIL: t-encrypt-sign.py
FAIL: t-sign.py
FAIL: t-signers.py
FAIL: t-decrypt.py
FAIL: t-verify.py
FAIL: t-decrypt-verify.py
FAIL: t-sig-notation.py
FAIL: t-export.py
FAIL: t-import.py
FAIL: t-trustlist.py
FAIL: t-edit.py
FAIL: t-keylist.py
FAIL: t-keylist-from-data.py
FAIL: t-wait.py
FAIL: t-encrypt-large.py
FAIL: t-file-name.py
FAIL: t-idiomatic.py
FAIL: t-protocol-assuan.py
FAIL: t-quick-key-creation.py
FAIL: t-quick-subkey-creation.py
FAIL: t-quick-key-manipulation.py
FAIL: t-quick-key-signing.py
FAIL: final.py
29 tests run, 0 succeeded, 29 failed, 0 skipped.
make[3]: *** [Makefile:613: xcheck] Error 29
make[3]: Leaving directory '/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/tests'
make[2]: *** [Makefile:472: check-recursive] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python'
make[1]: *** [Makefile:457: check-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang'
make: *** [Makefile:534: check-recursive] Error 1
builder for '/nix/store/3hvi5bzsjb0x1wrdhwqazs8akpqk5qa5-gpgme-1.13.1.drv' failed with exit code 2
cannot build derivation '/nix/store/napnjdcs66z1sagmndjhipv3pjm69yp3-python3.7-pgpy-0.5.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ncwjx5g23yhak4vvvq4qzmzgnkscclq2-python3.7-srht-0.54.2.drv': 1 dependencies couldn't be built
```
</details>

full log: https://logs.nix.ci/?key=nixos/nixpkgs.70990&attempt_id=d202832a-d134-4823-b199-2a1e24e7e4c4

### PR DESCRIPTION
**Pull request by kirelagin**
* Replace LD_LIBRARY_PATH with OS-specific name (e.g. DYLD_LIBRARY_PATH
  on macOS).
* Disable Python tests on macOS, because they use gpg, which fails due
  to a very long socket path (https://github.com/NixOS/nix/pull/1085).

The former should be fixed upstream ([patch](https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/034591.html)). The latter is a Nix-specific issue,
but it can be worked-around upstream by making Python tests respect
--disable-gpg-test.

Fixes #70996.
Related NixOS/nix#1085.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gpgme/default.nix [hunk 1]
```diff
@@ -49,6 +49,15 @@ stdenv.mkDerivation rec {
 
   postPatch =''
     substituteInPlace ./configure --replace /usr/bin/file ${file}/bin/file
+  ''
+  # Apply only on Darwin to save rebuilds on Linux
+     + lib.optionalString stdenv.isDarwin ''
+    sed -i "s/LD_LIBRARY_PATH/@shlibpath_var@/" lang/python/tests/Makefile.in
+    sed -i "s/ac_subst_vars='/ac_subst_vars='shlibpath_var\n/" configure
```
Comment (Mic92) [inline pkgs/development/libraries/gpgme/default.nix:56]: It would be nice to also send this one upstream: https://dev.gnupg.org/

Comment (Mic92) [reply pkgs/development/libraries/gpgme/default.nix:56]: cc @AndreHeinecke

Comment (kirelagin) [reply pkgs/development/libraries/gpgme/default.nix:56]: I sent it.

Comment (Mic92) [reply pkgs/development/libraries/gpgme/default.nix:56]: Sounds good. Can you provide a link?

Comment (kirelagin) [reply pkgs/development/libraries/gpgme/default.nix:56]: They accept patches by email, and, for some reason, it hasn’t appeared in https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/ yet.

Comment (kirelagin) [reply pkgs/development/libraries/gpgme/default.nix:56]: https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/034591.html

File: pkgs/development/libraries/gpgme/default.nix [hunk 2]
```diff
@@ -33,7 +34,14 @@ stdenv.mkDerivation rec {
...
     })
-  ];
+    # https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/034591.html
+    (fetchpatch {
+      name = "0001-Fix-python-tests-on-non-Linux.patch";
+      url = "https://lists.gnupg.org/pipermail/gnupg-devel/attachments/20200415/f7be62d1/attachment.obj";
+      sha256 = "00d4sxq63601lzdp2ha1i8fvybh7dzih4531jh8bx07fab3sw65g";
+    })
+    # Disable python tests on Darwin as they use gpg (see configureFlags below)
+  ] ++ lib.optional stdenv.isDarwin ./disable-python-tests.patch;
```
Comment (Mic92) [inline pkgs/development/libraries/gpgme/default.nix:44]: This way it is easier to maintain and less likely to accidentally to break on updates.

Comment (Mic92) [reply pkgs/development/libraries/gpgme/default.nix:44]: Please let me know if it still works.

Comment (kirelagin) [reply pkgs/development/libraries/gpgme/default.nix:44]: It works.

**Pull request merged**